### ISSUE TITLE
mkdir race condition inspection: "Harden" sometimes changes meaning of code

### ISSUE THREAD

**Issue by fredricj** [kalessil/phpinspectionsea#1035]
Language Level: PHP 7.2
EA Ultimate 2.0.8

When applying the fix for the following (allegedly badly written) code snippet:
```
$image_dir = "/tmp/test2";
if (!file_exists($image_dir)) {
	mkdir($image_dir, 0755, true) or die('Could not create directory!');
}
```
it changes to this:
```
$image_dir = "/tmp/test2";
if (!file_exists($image_dir)) {
	mkdir($image_dir, 0755, true) || mkdir($image_dir, 0755, true) || is_dir($image_dir);
}
````
Which have a completely different meaning

**Comment by kalessil** [comment]
Thank you for reporting @fredricj, looks like a false-positive. I'll take care about it.

**Comment by kalessil** [comment]
Fixed!
